pkg/operator/k8sutil/pod.go

```diff
@@ -114,3 +119,49 @@ func MakeRookImage(version string) string {
 func SetPodVersion(pod *v1.Pod, key, version string) {
 	pod.Annotations[key] = version
 }
+
+// DeleteDaemonDeployment makes a best effort at deleting the daemon pods, including the deployment and daemonset
+func DeleteDaemonDeployment(context *clusterd.Context, namespace, name string) {
+	logger.Infof("removing %s pods if they exist", name)
+
+	var gracePeriod int64
+	propagation := metav1.DeletePropagationForeground
+	options := &metav1.DeleteOptions{GracePeriodSeconds: &gracePeriod, PropagationPolicy: &propagation}
+
+	// Delete the deployment if it exists
+	err := context.Clientset.ExtensionsV1beta1().Deployments(namespace).Delete(name, options)
```

Comment:
this function is responsible for deleting the thing whether it is a daemonset or a deployment. not sure how we benefit from splitting it to two functions

Reply:
I find that DeleteDaemonDeployment()  seems very specific. Utils functions are better made to be generic so that they are more compact, simpler and better reusable. If you want, you can define a function like DeleteAll() which also includes pods, replicaset and replicationcontroller. That can be reuse as a clean up.

Reply:
when specific to rgw it made sense, but i see your point for a generic helper. i'll refactor